core/src/main/java/org/apache/spark/shuffle/sort/BypassMergeSortShuffleWriter.java

```diff
@@ -124,13 +153,19 @@ public void insertAll(Iterator<Product2<K, V>> records) throws IOException {
     for (DiskBlockObjectWriter writer : partitionWriters) {
       writer.commitAndClose();
     }
+
+    partitionLengths =
+      writePartitionedFile(shuffleBlockResolver.getDataFile(shuffleId, mapId));
+    shuffleBlockResolver.writeIndexFile(shuffleId, mapId, partitionLengths);
+    mapStatus = MapStatus$.MODULE$.apply(blockManager.shuffleServerId(), partitionLengths);
   }
 
-  @Override
-  public long[] writePartitionedFile(
-      BlockId blockId,
-      TaskContext context,
-      File outputFile) throws IOException {
+  // Exposed for testing
+  long[] getPartitionLengths() {
+    return partitionLengths;
+  }
+
+  private long[] writePartitionedFile(File outputFile) throws IOException {
```

Comment:
can u add inline doc explaining what long[] is (in particular explain whether its offset, or length)